packages/react-storage/src/components/StorageBrowser/composables/Pagination.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { CLASS_BASE } from '../views/constants';
+import {
+  ButtonElement,
+  ButtonElementProps,
+  IconElement,
+  ListItemElement,
+  NavElement,
+  OrderedListElement,
+  SpanElement,
+} from '../context/elements';
+import { DataListProps } from '../components/types';
+
+export interface PaginationProps {
+  currentPage: number;
+  disableNext: boolean;
+  disablePrevious: boolean;
+  handlePaginateNext: () => void;
+  handlePaginatePrevious: () => void;
+}
+
+export type PaginationVariant = `paginate-${'next' | 'current' | 'previous'}`;
+
+const BLOCK_NAME = `${CLASS_BASE}__pagination`;
+const DEFAULT_VARIANTS: PaginationVariant[] = [
+  'paginate-previous',
+  'paginate-current',
+  'paginate-next',
+];
+
+const getButtonVariantProps = (
```

Comment:
Can we simplify all this a bit and just create an array of button items directly? It is our own pagination composable and it seems unlikely we'll be doing something like having multiple buttons that all do "paginate-next"

Reply:
I changed this to a direct array of button props. Let me know what you think